From 62dc75474acc0c0eb149a31ddc88e80f49bddf3b Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <cyanogenmod@cerqueira.org>
Date: Sat, 1 Dec 2012 18:38:53 +0000
Subject: [PATCH 3/5] init: Bring back support for arbitrary chargermode
 cmdlines

This was accidentally removed while debugging the init breakage
fixed in change I8ef5c932efcd5e7f8d6f6fce0915683d84c3ee11. Put
it back

Change-Id: Iaacc30abff0bcd9fa49f98b158b293fa363d9ea3
---
 init/init.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/init/init.cpp b/init/init.cpp
index da8ca66..5cd0939 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -83,7 +83,13 @@ struct selabel_handle *sehandle_prop;
 
 static int property_triggers_enabled = 0;
 
+#ifndef BOARD_CHARGING_CMDLINE_NAME
+#define BOARD_CHARGING_CMDLINE_NAME "androidboot.battchg_pause"
+#define BOARD_CHARGING_CMDLINE_VALUE "true"
+#endif
+
 static char qemu[32];
+static char battchg_pause[32];
 
 std::string default_console = "/dev/console";
 static time_t process_needs_restart_at;
@@ -481,6 +487,8 @@ static void import_kernel_nv(const std::string& key, const std::string& value, b
 
     if (key == "qemu") {
         strlcpy(qemu, value.c_str(), sizeof(qemu));
+    } else if (key == BOARD_CHARGING_CMDLINE_NAME) {
+        strlcpy(battchg_pause, value.c_str(), sizeof(battchg_pause));
     } else if (android::base::StartsWith(key, "androidboot.")) {
         property_set("ro.boot." + key.substr(12), value);
     }
@@ -1192,7 +1200,8 @@ int main(int argc, char** argv) {
 
     // Don't mount filesystems or start core system services in charger mode.
     std::string bootmode = GetProperty("ro.bootmode", "");
-    if (bootmode == "charger" || charging_mode_booting()) {
+    if (bootmode == "charger" || charging_mode_booting() ||
+            strcmp(battchg_pause, BOARD_CHARGING_CMDLINE_VALUE) == 0) {
         am.QueueEventTrigger("charger");
     } else {
         am.QueueEventTrigger("late-init");
-- 
2.7.4

